substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs

```diff
@@ -0,0 +1,218 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! API implementation for broadcasting transactions.
+
+use crate::{transaction::api::TransactionBroadcastApiServer, SubscriptionTaskExecutor};
+use codec::Decode;
+use futures::{FutureExt, StreamExt};
+use futures_util::stream::AbortHandle;
+use jsonrpsee::core::{async_trait, RpcResult};
+use parking_lot::RwLock;
+use rand::{distributions::Alphanumeric, Rng};
+use sc_client_api::BlockchainEvents;
+use sc_transaction_pool_api::{
+	BlockHash, TransactionFor, TransactionPool, TransactionSource, TransactionStatus,
+};
+use sp_blockchain::HeaderBackend;
+use sp_core::Bytes;
+use sp_runtime::traits::Block as BlockT;
+use std::{collections::HashMap, sync::Arc};
+
+use super::error::ErrorBroadcast;
+
+/// An API for transaction RPC calls.
+pub struct TransactionBroadcast<Pool, Client> {
+	/// Substrate client.
+	client: Arc<Client>,
+	/// Transactions pool.
+	pool: Arc<Pool>,
+	/// Executor to spawn subscriptions.
+	executor: SubscriptionTaskExecutor,
+	/// The brodcast operation IDs.
+	broadcast_ids: Arc<RwLock<HashMap<String, BroadcastState>>>,
+}
+
+/// The state of a broadcast operation.
+struct BroadcastState {
+	/// Handle to abort the running future that broadcasts the transaction.
+	handle: AbortHandle,
+}
+
+impl<Pool, Client> TransactionBroadcast<Pool, Client> {
+	/// Creates a new [`TransactionBroadcast`].
+	pub fn new(client: Arc<Client>, pool: Arc<Pool>, executor: SubscriptionTaskExecutor) -> Self {
+		TransactionBroadcast { client, pool, executor, broadcast_ids: Default::default() }
+	}
+
+	/// Generate an unique operation ID for the `transaction_broadcast` RPC method.
+	pub fn generate_unique_id(&self) -> String {
+		let generate_operation_id = || {
+			// The lenght of the operation ID.
+			const OPERATION_ID_LEN: usize = 16;
+
+			let mut rng = rand::thread_rng();
+			(&mut rng)
```

Comment:
I can see that this is documented in `rand` because `sample_iter` takes `self`, but since you don't use `rng` again after this line, perhaps you don't need it?

Reply:
Thanks for the suggestion! Looks much cleaner now